lang/types/value.go

```diff
@@ -58,9 +58,20 @@ func ValueOf(v reflect.Value) (Value, error) {
 	typ := value.Type()
 	kind := typ.Kind()
 	for kind == reflect.Ptr {
+		// Not much that can be done here. Caller must pass a Settable Ptr value.
+		if !value.CanSet() {
```

Comment:
I'm not 100% sure this is what you want. In situations like these, what I'd recommend doing, is adding a single simple test with the short, standalone bit that you want to test:

Like:
TestFrebibNewTest(t....) ...  {
v, err := ValueOf(XXX)
if err != nil ...
if v == nil ...
if err := v.Cmp(XXX); err != nil ...
}

I think that would help us better understand where you're stuck, and then we can go through and understand the reflect lib and which method you want a bit better.

Reply:
You're right. I was seriously overthinking this. Reworked it to a really simple change. Most of the diff is just adjusting the tests now.